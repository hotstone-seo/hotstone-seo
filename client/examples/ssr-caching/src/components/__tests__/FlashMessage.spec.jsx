import React from 'react';
import FlashMessage from '../FlashMessage.jsx';

test('Should render FlashMessage correctly', () => {
  const resetFlash = jest.fn();
  const flash = {
    show: true,
    type: 'error',
    text: 'test'
  };
  const tree = enzyme.mount(
    <FlashMessage.WrappedComponent flash={flash} resetFlash={resetFlash} />
  );

  expect(tree).toMatchSnapshot();

  tree.setProps({
    flash: {
      ...flash,
      text: 'test2'
    }
  });
  tree.setProps({
    flash: {
      ...flash,
      show: false
    }
  });
});
