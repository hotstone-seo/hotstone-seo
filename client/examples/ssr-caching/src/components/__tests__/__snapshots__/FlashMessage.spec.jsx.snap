// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FlashMessage correctly 1`] = `
<FlashMessage
  flash={
    Object {
      "show": true,
      "text": "test",
      "type": "error",
    }
  }
  resetFlash={[MockFunction]}
>
  <div
    className="flash error"
  >
    test
  </div>
</FlashMessage>
`;
